content/en/apps/guides/hosting/4.x/data-migration.md

```diff
@@ -0,0 +1,270 @@
+```shell
+mkdir -p ~/couchdb-migration/ 
+cd ~/couchdb-migration/ 
+curl -s -o ./docker-compose.yml https://raw.githubusercontent.com/medic/couchdb-migration/main/docker-compose.yml
+docker-compose up
+```
+
+For the following steps, the tool needs access to your CouchDb installation. To allow this access, you will need to provide a URL to your CouchDB installation that includes authentication. 
+Additionally, if your CouchDb runs in docker, the tool needs to be added to the same docker network in order to access protected endpoints:
+
+```shell
+export COUCH_URL=http://<authentication>@<host-ip>:<port>
+```
+or 
+```shell
+export CHT_NETWORK=<docker-network-name>
+export COUCH_URL=http://<authentication>@<docker-service-name>:<port>
+```
+
+For simplicity, you could store these required values in an `.env` file: 
+```shell
+cat > ${HOME}/couchdb-migration/.env << EOF
+CHT_NETWORK=<docker-network-name>
+COUCH_URL=http://<authentication>@<docker-service-name>:<port>
+EOF
+```
+
+### 2. Prepare CHT-Core 3.x installation for upgrading
+
+Backup your data! If you encounter any problems executing the instructions of this guide, you should be able to restore your CHT 3X instance using the backup data.
+[Consult information about backups for details]({{< relref "apps/guides/hosting/3.x/self-hosting#backup" >}}).
+To ensure no changes happen to your CouchDB data in your CHT 3.x server, stop the API by getting shell on your Medic OS container and calling `/boot/svc-stop medic-api`.
+
+To minimize downtime when upgrading, it's advised to prepare the 3.x installation for the 4.x upgrade, and pre-index all views that are required by 4.x.
+
+The migration tool provides a command which will download all 4.x views to your 3.x CouchDb, and initiate view indexing. `<desired CHT version>` is any version at or above `4.0.0`:
+
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration pre-index-views <desired CHT version>
+```
+
+Once view indexing is finished, proceed with the next step.
+
+{{% alert title="Note" %}} If this step is omitted, 4.x API will fail to respond to requests until all views are indexed. Depending on the size of the database, this could take many hours, or even days. {{% /alert %}} 
+
+### 3. Save existent CouchDb configuration
+
+Some CouchDb configuration values must be ported from existent CouchDb to the 4.x installation. Store them in a safe location before shutting down 3.x CouchDb.
+Use the migration tool to obtain these values:
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration get-env
+```
+
+##### a) CouchDB secret 
+Used in encrypting all CouchDb passwords and session tokens.
+##### b) CouchDb server uuid
+Used in generating replication checkpointer documents, which track where replication progress between every client and the server, and ensure that clients don't re-download or re-upload documents.
+
```

Comment:
Having run through the guide now once (almost!), I think we can be a bit more explicit here:

```suggestion
Some CouchDb configuration values must be ported from existent CouchDb to the 4.x installation. Store them in a safe location before shutting down 3.x CouchDb.
Use the migration tool to obtain these values:
```shell
cd ~/couchdb-migration/ 
docker-compose run couch-migration get-env
```

Four values will be output. You already know the user (`COUCHDB_USER`) and password (`COUCHDB_PASSWORD`), but the two you need to save for later are:

* `COUCHDB_SECRET` -  CouchDB secret - Used in encrypting all CouchDB passwords and session tokens.
* `COUCHDB_UUID` -  `CouchDB server UUID - Used in generating replication checkpointer documents, which track where replication progress between every client and the server, and ensure that clients don't re-download or re-upload documents.
```